support/config/deployment.go

```diff
@@ -81,13 +81,17 @@ func (c *DeploymentConfig) ApplyTo(deployment *appsv1.Deployment) {
 	} else {
 		deployment.Spec.Replicas = pointer.Int32(int32(c.Replicas))
 	}
-	// there are two standard cases currently with hypershift: HA mode where there are 3 replicas spread across
-	// zones and then non ha with one replica. When only 3 zones are available you need to be able to set maxUnavailable
-	// in order to progress the rollout. However, you do not want to set that in the single replica case because it will
-	// result in downtime.
+	// there are three standard cases currently with hypershift: HA mode where there are 3 replicas spread across
+	// zones, HA mode with 2 replicas, and then non ha with one replica. When only 3 zones are available you need
+	// to be able to set maxUnavailable in order to progress the rollout. However, you do not want to set that in
+	// the single replica case because it will result in downtime.
 	if c.Replicas > 1 {
-		maxSurge := intstr.FromInt(0)
-		maxUnavailable := intstr.FromInt(1)
+		maxSurge := intstr.FromInt(1)
+		maxUnavailable := intstr.FromInt(0)
```

Comment:
`maxSurge:1` and `maxUnavailable: 0` does not work in the `TestCreateClusterRequestServingIsolation` case because `oauth-openshift` runs on the request serving nodes, of which there are only two.  Thus surging is not possible.

Thanks for @csrwng for figuring this out!

Reply:
I've updated the PR to set `maxUnavailable: 1` for oauth. We should still allow `maxSurge: 1` in the event there are more serving nodes available.